internal/store/engine_state.go

```diff
@@ -614,16 +614,34 @@ func (ms *ManifestState) HasPendingChangesBeforeOrEqual(highWaterMark time.Time)
 var _ model.TargetStatus = &ManifestState{}
 
 func ManifestTargetEndpoints(mt *ManifestTarget) (endpoints []model.Link) {
+	// TODO(maia): we should probably let users define the order of their endpoints;
+	//  at minimum, arbitrary links and port fwd URLs should stay together (currently
+	//  if a k8s resource has both links and port fwds, the whole list of endpoints
+	//  will be alphabetized). However, we don't expect lots of people to specify
+	//  links and port fwds on the same resource, so this is fine for now.
```

Comment:
> However, we don't expect lots of people to specify links and port fwds on the same resource, so this is fine for now.

Hmm, I have not talked to any users about this, but I'd expect by far the most common use case to be to point at paths *on* the port forward, and for it to be rare to use links without port forwards.

I suspect a nicer default would be to use this order:
1. arbitrary links, in order specified in the tiltfile
2. port forwards

(1) gives the user mostly total control, and if they really want to move a port forward link up, they can specify it also as an arbitrary link.

If most people are not specifying links and port forwards on the same resource, it seems like it could be annoying of us to take their lovingly ordered links and force them into sorted order.

Reply:
> Hmm, I have not talked to any users about this, but I'd expect by far the most common use case to be to point at paths on the port forward, and for it to be rare to use links without port forwards.

That's a use case for the `path` arg introduced in #3858 -- imho you should use that instead of specifying port forward 8000 and link "localhost:8000/v1/ui". (Tho you're probs right that we'll get ppl doing that anyway.)

I agree that the ordering is dumb tho--let me tackle that in a separate PR?